src/thor/bidirectional_astar.cc

```diff
@@ -230,17 +230,26 @@ inline bool BidirectionalAStar::ExpandInner(baldr::GraphReader& graphreader,
     }
   }
 
-  // Get cost. Separate out transition cost.
-  sif::Cost transition_cost =
-      FORWARD ? costing_->TransitionCost(meta.edge, nodeinfo, pred)
-              : costing_->TransitionCostReverse(meta.edge->localedgeidx(), nodeinfo, opp_edge,
-                                                opp_pred_edge, pred.has_measured_speed(),
-                                                pred.internal_turn());
+  // Get cost
   uint8_t flow_sources;
   sif::Cost newcost =
-      pred.cost() + transition_cost +
-      (FORWARD ? costing_->EdgeCost(meta.edge, tile, time_info.second_of_week, flow_sources)
-               : costing_->EdgeCost(opp_edge, t2, time_info.second_of_week, flow_sources));
+      pred.cost() + (FORWARD
+                         ? costing_->EdgeCost(meta.edge, tile, time_info.second_of_week, flow_sources)
+                         : costing_->EdgeCost(opp_edge, t2, time_info.second_of_week, flow_sources));
+
+  // Separate out transition cost.
+  InternalTurn internal_turn;
+  sif::Cost transition_cost;
+  if (FORWARD) {
+    transition_cost = costing_->TransitionCost(meta.edge, nodeinfo, pred);
+  } else {
+    internal_turn = costing_->TurnType(meta.edge->localedgeidx(), nodeinfo, opp_edge, opp_pred_edge);
```

Comment:
`TurnType` method doesn't look like a trivial operation so I decided to memoize its result and reuse later when adding new label